pkgs/applications/editors/emacs/elisp-packages/manual-packages/tsc/package.nix

```diff
@@ -1,70 +1,50 @@
 { lib
-, symlinkJoin
 , melpaBuild
 , fetchFromGitHub
 , rustPlatform
-
-, runtimeShell
-, writeScript
-, python3
-, nix-prefetch-github
-, nix
+, stdenv
 }:
 
 let
+  libExt = stdenv.hostPlatform.extensions.sharedLibrary;
 
-  srcMeta = lib.importJSON ./src.json;
-  inherit (srcMeta) version;
-
-  src = fetchFromGitHub srcMeta.src;
-
-  tsc = melpaBuild {
-    inherit src;
-    inherit version;
-
-    pname = "tsc";
-
-    sourceRoot = "${src.name}/core";
-  };
-
-  tsc-dyn = rustPlatform.buildRustPackage {
-    inherit version;
-    inherit src;
-
+  tsc-dyn = rustPlatform.buildRustPackage rec {
     pname = "tsc-dyn";
+    version = "0.18.0";
+
+    src = fetchFromGitHub {
+      owner = "emacs-tree-sitter";
+      repo = "emacs-tree-sitter";
+      rev = "909717c685ff5a2327fa2ca8fb8a25216129361c";
+      hash = "sha256-LrakDpP3ZhRQqz47dPcyoQnu5lROdaNlxGaQfQT6u+k=";
+    };
+
+    cargoLock = {
+      lockFile = ./Cargo.lock;
+      outputHashes = {
+        "tree-sitter-0.20.0" = "sha256-hGiJZFrQpO+xHXosbEKV2k64e2D8auNGEtdrFk2SsOU=";
+      };
+    };
 
-    nativeBuildInputs = [ rustPlatform.bindgenHook ];
     sourceRoot = "${src.name}/core";
 
     postInstall = ''
-      LIB=($out/lib/libtsc_dyn.*)
-      TSC_PATH=$out/share/emacs/site-lisp/elpa/tsc-${version}
-      install -d $TSC_PATH
-      install -m444 $out/lib/libtsc_dyn.* $TSC_PATH/''${LIB/*libtsc_/tsc-}
-      echo -n $version > $TSC_PATH/DYN-VERSION
-      rm -r $out/lib
+      pushd $out/lib
+      mv --verbose libtsc_dyn${libExt} tsc-dyn${libExt}
+      echo -n $version > DYN-VERSION
+      popd
     '';
-
-    inherit (srcMeta) cargoHash;
   };
+in melpaBuild {
+  pname = "tsc";
+  inherit (tsc-dyn) version src;
+
+  files = ''("core/*.el" "${tsc-dyn}/lib/*")'';
 
-in symlinkJoin {
-  name = "tsc-${version}";
-  paths = [ tsc tsc-dyn ];
+  ignoreCompilationError = false;
 
   passthru = {
-    updateScript = let
```

Comment:
No existing update scripts work for a rust package using `cargoLock.lockFile`, right?

Reply:
It turns out `nix-update-script` works.  I have updated the PR to use that as the update script.